src/Network/Probe.php

```diff
@@ -333,6 +333,10 @@ public static function lrdd(string $uri)
 	 */
 	public static function uri($uri, $network = '', $uid = -1)
 	{
+		// Convert @user@domain.tld format to the expected user@domain.tld format
```

Comment:
I don't think that we need this conversion here. We can assume that the input is always sane. When we do some cleaning then I would suggest doing this in the frontend. This function here receives addresses in various formats (`user@domain.tld`, `https://domain.tld/user`, `https://domain.tld/@user`, ...)

Reply:
We don't need it but it's practical since it doesn't affect anything else than `@user@domain.tld`.

Reply:
This function is not only used to fetch information for a given address. it also is used to check if some address is valid. I'm not totally sure, but theoretically this change could lead to a situation where some invalid data (with `@user@domain.tld`) is stored into the contact (or apcontact) table because it seems to be valid data.

Reply:
Hrm.